miprometheus/utils/sampler_factory.py

```diff
@@ -0,0 +1,169 @@
+
+                indices = params['indices']
+
+                # Analyze the type.
+                if type(indices) == str:
+
+                    # Try to open the file.
+                    try:
+                        # from expanduser()'s doc: If the expansion fails or if the path does not begin
+                        # with a tilde, the path is returned unchanged. -> So operation below should be safe.
+                        file = open(os.path.expanduser(indices), "r")
+                        # Read the file.
+                        indices = file.readline() 
+                        # Truncate the last "enter"
+                        indices = indices[:-1]
+                        file.close()
+
+                    except Exception:
+                        # Ok, this is not a file.
+                        pass
+
+                    # If indices are already square brackets [].
+                    if indices[0] == '[' and indices[-1] == ']':
+                        # Remove the brackets.
+                        indices = indices.replace("[", "").replace("]", "")
+
+                    # Get digits.
+                    digits = indices.split()
+                    if len(digits) == 2:
+                        # Create simple range.
+                        indices = range(int(digits[0]), int(digits[1]))
+                    else:
+                        # Use them as a list.
+                        indices = [int(x) for x in digits]
+
+                # Check if there aren't too many indices.
+                # if len(indices) > len(problem):
+                #    logger.error("Length of indices if greater than the number of samples in the problem!")
+                #    exit(-1)
+
+                # Check if indices are within range.
+                if max(indices) >= len(problem):
+                    logger.error("SubsetRandomSampler cannot work properly when indices are out of range ({}) "
+                                 "considering that there are {} samples in the problem!".format(max(indices),
+                                                                                                len(problem)))
+                    exit(-2)
+                sampler = sampler_class(indices)
+            else:
+                # Create "regular" sampler.
+                sampler = sampler_class(problem)
+
+            # Return sampler.
+            return sampler
+        except Exception as e:
+            logger.error(e)
+            logger.warning("Using default sampling without sampler.")
+            return None
+
+
+if __name__ == "__main__":
+    """
+    Tests the factory.
+    """
+    from miprometheus.utils.param_interface import ParamInterface
+
+    # Problem.
+    class TestProblem(object):
+        def __len__(self):
+            return 50
+
+    # All samplers operate on TestProblem only,
+    # whereas SubsetRandomSampler additionally accepts 'indices' with three options.
```

Comment:
Note: `torch.utils.data.WeightedRandomSampler`, `torch.utils.data.BatchSampler` & `torch.utils.data.distributed.DistributedSampler` are more complex than the other `Samplers` (including `SubsetRandomSampler` and the factory fails if trying to build them: 

![error](https://user-images.githubusercontent.com/33840936/47576330-065cc700-d8f9-11e8-85df-048d41c89c04.png)

So if we do not want to put in the effort to handle these `Samplers` right now, which I totally understand because of the time constraints, let's just warn the user and exit the factory.

Reply:
good catch